nmigen_boards/arty_z7.py

```diff
@@ -0,0 +1,175 @@
+    device      = "xc7z020"
+    package     = "clg400"
+    speed       = "1"
+    default_clk = "clk125"
+    resources   = [
+        Resource("clk125", 0,
+            Pins("H16", dir="i"), Clock(125e6), Attrs(IOSTANDARD="LVCMOS33")),
+
+        *SwitchResources(
+            pins="M20 M19",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        RGBLEDResource(0,
+            r="N15", g="G17", b="L15",                          # LD4
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        RGBLEDResource(1,                                       # LD5
+            r="M15", g="L14", b="G14",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        *LEDResources(
+            pins="R14 P14 N16 M14",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        *ButtonResources(
+            pins="D19 D20 L20 L19",
+            attrs=Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("audio", 0,
+            Subsignal("pwm", Pins("R18", dir="o")),
+            Subsignal("sd",  PinsN("T17", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("crypto_sda", 0,                               # ATSHA204A
+            Pins("J15", dir="io"),
+            Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("hdmi_rx", 0,                                  # J10
+            Subsignal("cec", Pins("H17", dir="io")),
+            Subsignal("clk", DiffPairs("N18", "P19", dir="i"),
+                Attrs(IO_TYPE="TMDS_33")),
+            Subsignal("d",   DiffPairs("V20 T20 N20", "W20 U20 P20", dir="i"),
+                Attrs(IO_TYPE="TMDS_33")),
+            Subsignal("hpd", Pins("T19", dir="o")),
+            Subsignal("scl", Pins("U14", dir="io")),
+            Subsignal("sda", Pins("U15", dir="io")),
+            Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("hdmi_tx", 0,                                  # J11
+            Subsignal("cec", Pins("G15", dir="io")),
+            Subsignal("clk", DiffPairs("L16", "L17", dir="o"),
+                Attrs(IO_TYPE="TMDS_33")),
+            Subsignal("d",   DiffPairs("K17 K19 J18", "K18 J19 H18", dir="o"),
+                Attrs(IO_TYPE="TMDS_33")),
+            Subsignal("hpd", PinsN("R19", dir="i")),
+            Subsignal("scl", Pins("M17", dir="io")),
+            Subsignal("sda", Pins("M18", dir="io")),
+            Attrs(IOSTANDARD="LVCMOS33"))
+    ]
+    connectors = [
+        Connector("pmod", 0, "Y18 Y19 Y16 Y17 - - U18 U19 W18 W19 - -"),  # JA
+        Connector("pmod", 1, "Y14 W14 T10 T11 - - W16 V16 W13 V12 - -"),  # JB
+
+        Connector("ck_io", 0, {
+            # Outer Digital Header
+            "io0": "T14",
+            "io1": "U12",
+            "io2": "U13",
+            "io3": "V13",
+            "io4": "V15",
+            "io5": "T15",
+            "io6": "R16",
+            "io7": "U17",
+            "io8": "V17",
+            "io9": "V18",
+            "io10": "T16",
+            "io11": "R17",
+            "io12": "P18",
+            "io13": "N17",
```

Comment:
Are you sure this is the best numbering scheme for this connector? You're the one who is going to be using the board, so I assume it is more likely than not, just double-checking.

Reply:
No i'm not totally sure but it is very similar to the Arty A7 so I just stick to the existing numbering scheme for this existing board (which is in fact as printed on the silkscreen, so at least it is unambiguous)